src/i18n/rsc-trans.tsx

```diff
@@ -1,14 +1,16 @@
 import React from "react"
 
-import { getI18n } from './i18n';
+import { useLinguiSSR} from './i18n';
 import { TransNoContext, TransProps } from '@lingui/react/server';
 
-export function Trans(props: TransProps): React.ReactElement<any, any> | null {
-  const i18n = getI18n()
+
+
+export async function Trans(props: TransProps): Promise<React.ReactElement<any, any> | null> {
+  const i18n = await useLinguiSSR();
```

Comment:
if somehow we could get the locale here, it wouldn't be necesary to `await useLinguiSSR(params.locale)` on every page.